js/ui/windowManager.js

```diff
@@ -1423,6 +1423,13 @@ const WindowManager = new Lang.Class({
                         return;
                     let rect = currentSession[0].windowBuilder.get_frame_rect();
                     currentSession[0].buttonBuilder.set_position(rect.x + rect.width - 100, rect.y + rect.height - 100);
+                    if (currentSession[0].windowApp === undefined)
+                        return;
+                    currentSession[0].windowApp.move_resize_frame(false,
+                                                                  rect.x,
+                                                                  rect.y,
+                                                                  rect.width,
+                                                                  rect.height);
```

Comment:
If so this could be separated out into a separate method `moveResizeCorrespondingWindowApp`